File: app/src/main/java/com/agus/project/dicodingtest/view/PlayersDetailActivity.java
package com.agus.project.dicodingtest.view;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;
import android.widget.TextView;

import com.agus.project.dicodingtest.R;
import com.bumptech.glide.Glide;

import butterknife.BindView;
import butterknife.ButterKnife;

public class PlayersDetailActivity extends AppCompatActivity {

    public static final String _NAME = "NAME";
    public static final String _TEAM = "TEAM";
    public static final String _DATE = "DATE";
    public static final String _WAGE = "WAGE";
    public static final String _POS = "POS";
    public static final String _DESC = "DESC";
    public static final String _IMAGE= "IMAGE";


    @BindView(R.id.player_name)
    TextView playerName;

    @BindView(R.id.player_club)
    TextView playerTeam;

    @BindView(R.id.player_date_born_)
    TextView dateBorn;

    @BindView(R.id.player_wage)
    TextView playerWage;

    @BindView(R.id.player_position)
    TextView playerPosition;

    @BindView(R.id.player_description)
    TextView playerDesc;

    @BindView(R.id.player_image)
    ImageView playerImage;



    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_player_detail);
        ButterKnife.bind(this);

        if(getIntent().getExtras()!=null){
            setData(getIntent().getExtras());
        }

    }

    private void setData(Bundle bundle){
        playerName.setText(bundle.getString(_NAME));
        playerTeam.setText(bundle.getString(_TEAM));
        dateBorn.setText(bundle.getString(_DATE));
        playerWage.setText(bundle.getString(_WAGE));
        playerPosition.setText(bundle.getString(_POS));
        playerDesc.setText(bundle.getString(_DESC));
        Glide.with(this).load(bundle.getString(_IMAGE)).into(playerImage);
    }
}

File: app/src/main/java/com/agus/project/dicodingtest/view/PlayersDetailActivity.kt
package com.agus.project.dicodingtest.view

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.ImageView
import android.widget.TextView

import com.agus.project.dicodingtest.R
import com.bumptech.glide.Glide

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.Date

import butterknife.BindView
import butterknife.ButterKnife
import kotlinx.android.synthetic.main.activity_player_detail.*
import kotlinx.android.synthetic.main.players_content.*
import kotlinx.android.synthetic.main.players_content.player_club
import kotlinx.android.synthetic.main.players_content.player_name

class PlayersDetailActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_player_detail)
        ButterKnife.bind(this)

        if (intent.extras != null) {
            setData(intent.extras!!)
        }

        back_btn.setOnClickListener { v ->
            startActivity(Intent(this,MainActivity::class.java))
        }

    }

    private fun setData(bundle: Bundle) {
        player_name!!.text = bundle.getString(_NAME)
        player_club!!.text = bundle.getString(_TEAM)
        player_date_born_!!.text = newFormatDate(bundle.getString(_DATE))
        player_wage!!.text = bundle.getString(_WAGE)
        player_pos!!.text = bundle.getString(_POS)
        player_description!!.text = bundle.getString(_DESC)
        Glide.with(this).load(bundle.getString(_IMAGE)).into(player_images!!)
    }

    private fun newFormatDate(oldDate: String?): String {
        val df = SimpleDateFormat("yyyy-MM-dd")
        val dfNew = SimpleDateFormat("dd MMMM yyyy")

        var date: Date? = null
        try {
            date = df.parse(oldDate)
        } catch (e: ParseException) {
            e.printStackTrace()
        }

        return dfNew.format(date)
    }

    companion object {

        val _NAME = "NAME"
        val _TEAM = "TEAM"
        val _DATE = "DATE"
        val _WAGE = "WAGE"
        val _POS = "POS"
        val _DESC = "DESC"
        val _IMAGE = "IMAGE"
    }

    override fun onBackPressed() {
        startActivity(Intent(this,MainActivity::class.java))
    }
}
